### PR TITLE
[webview_flutter] Enable setAllowFileAccess on Android setting when loading files

### PR DESCRIPTION
**Pull request by mvanbeusekom**
This makes two changes:

1. It adds the `WebSettings.setAllowFileAccess()` method to provide the option to enable/ disable this feature on Android;
2. Ensures that the `WebSettings.setAllowFileAccess()` is called from the `WebViewAndroidWidget.loadFile()` method to ensure this option is enabled. Before Android API 30 this option is enabled by default, however on Android API 30 and up this behaviour is flipped and the option is now disabled by default which means the `loadFile` method doesn't work on these Android versions.  This bug was not detected before because the webview_flutter_android example app is configured to target Android SDK 28 (details can be found here: https://developer.android.com/reference/android/webkit/WebSettings#setAllowFileAccess(boolean)).

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran [the auto-formatter]. (Unlike the flutter/flutter repo, the flutter/plugins repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

### CODE REVIEW

**Review by stuartmorgan — APPROVED** [review]
LGTM with the format issues fixed. I was wondering when closing that other PR last night how this worked 🙂 I missed the fact that the example was using an older version than the switch though.

I don't love the automatic permanent change of security setting when loading a file, but I didn't have a suggestion that doesn't feel more annoying for clients. We should do a follow-up to document that side effect in the app-facing method fix comment though.

File: packages/webview_flutter/webview_flutter_android/android/src/main/java/io/flutter/plugins/webviewflutter/GeneratedAndroidWebView.java [hunk 1]
```diff
@@ -1,7 +1,3 @@
-// Copyright 2013 The Flutter Authors. All rights reserved.
-// Use of this source code is governed by a BSD-style license that can be
-// found in the LICENSE file.
```
Comment (stuartmorgan) [inline packages/webview_flutter/webview_flutter_android/android/src/main/java/io/flutter/plugins/webviewflutter/GeneratedAndroidWebView.java:3]: This still needs to be re-added for CI to pass.

Comment (mvanbeusekom) [reply packages/webview_flutter/webview_flutter_android/android/src/main/java/io/flutter/plugins/webviewflutter/GeneratedAndroidWebView.java]: Sorry completely missed that. Just submitted an update.

**Comment by stuartmorgan** [comment]
Merging in master will help this pass Android tests

**Pull request merged**